pkg/cloudprovider/aws/object_storage_adapter.go

```diff
@@ -104,12 +109,17 @@ func (op *objectStorageAdapter) GetObject(bucket string, key string) (io.ReadClo
 	return res.Body, nil
 }
 
-func (op *objectStorageAdapter) ListCommonPrefixes(bucket string, delimiter string) ([]string, error) {
+func (op *objectStorageAdapter) ListCommonPrefixes(bucket string, delimiter string, prefix string) ([]string, error) {
+	op.logger.Infof("list s3 common prefix s3://%s/%s", bucket, prefix)
 	req := &s3.ListObjectsV2Input{
 		Bucket:    &bucket,
 		Delimiter: &delimiter,
 	}
 
+	if prefix != "" {
```

Comment:
What happens if we always specify `&prefix` regardless of its value? Does it still work?

Reply:
yeah, as long it's a valid path value. If it's not specified ("") then you get the old behavior anything else it will try to join the path and will use it.